sdk/servicebus/service-bus/test/deadLetter.spec.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Why not re-use the existing test cases that dead letter a message and then receive from the dead letter queue to verify it? Look for test cases with the below titles

- deadLetter() moves message to deadletter queue
- Deadlettering a deferred message moves it to dead letter queue

Reply:
I did see those and I still consider those good tests for the dead letter flow.

These tests are here because, despite the fact that seem to have quite a few tests, none of them covered these cases and I wanted it to be a bit more obvious.

Reply:
My concern is around having the similar concepts having tests in different places. For example, the minute I see a file with the name `deadLetter.spec.ts` I'd expect all tests related to dead lettering to be live here.

Reply:
Let me think about this for another PR. You've got a good point but our test suite isn't written in a way to be very granular, which can make wholesale reorganization a little annoying.